cpp/open3d/pipelines/registration/TransformationEstimation.cpp

```diff
@@ -89,6 +109,29 @@ Eigen::Matrix4d TransformationEstimationPointToPlane::ComputeTransformation(
     return is_success ? extrinsic : Eigen::Matrix4d::Identity();
 }
 
+std::tuple<std::shared_ptr<const geometry::PointCloud>,
+           std::shared_ptr<const geometry::PointCloud>>
+TransformationEstimationPointToPlane::InitializePointCloudsForTransformation(
+        const geometry::PointCloud &source,
+        const geometry::PointCloud &target,
+        double max_correspondence_distance) const {
+    if (!target.HasNormals()) {
+        utility::LogError(
+                "PointToPlaneICP requires target pointcloud to have normals.");
+    }
+    std::shared_ptr<const geometry::PointCloud> source_initialized_c(
+            &source, [](const geometry::PointCloud *) {});
```

Comment:
Thanks for the PR @nicolaloi.  Can you explain why the deleter is empty here?

Reply:
@benjaminum In some cases, like for PointToPoint and PointToPlane ICP, no actual modification/initialization of the point clouds is required. However, now there is a consistent interface across different ICP algorithms, so even in these cases the initialization function is being called, which is returning *shared pointers to const point clouds*. Since in these cases the point clouds remain unchanged during the initialization step, creating a costly copy would be unnecessary (eg *pc_initialized = make_shared<PointCloud>(pc_original)*). Instead, I’m creating the shared pointers from the addresses of the original point clouds, effectively creating a const "view" of the originals. The empty deleter ensures that when these shared pointers go out of scope, they don't free the memory owned by the original point clouds.

Reply:
OK, got it.